### ISSUE TITLE
Syntactic sugar for instantiating Components in tests

### ISSUE THREAD

**Issue by mhevery** [angular/angular#615]
Prerequisite: #614, #596 

```
describe('MyComponent', () => {
  it('should instantiate', @Injectable (tb:Testbed) => {
    var element:Element = tb
      .directive(MyDirective, { override: [new Template(...), new Component(...)] })
      .compile({
        html: '<my-component a="b" [c]="exp">lightdom</my-component>')
      .instantiate({some instance or Type})
      .element;
    });
  });
});

@Component({
  selector: 'my-component'
})
class MyComponent{}
```

**Comment by vicb** [comment]
Some remarks:

Traceur does not support types for arrow functions:
`(tb:Testbed) => null` does not compile, `function(tb:Testbed) { return null;}` is ok

Reflexion information are not added to anonymous functions:
If you have `it(name, function(tb:Testbed) {...})` then you can not retrieve the reflexion info on the anonymous function

**Comment by vsavkin** [comment]
Since testing is an important part of the Angular story, it'd be great to see a design of the test infrastructure before we start implementing it. 

For example, how do we support the following use cases:

- Running tests synchronously using MockZone
- Running tests asynchronously without mocking the zones api
- Testing components in isolation (without html)
- Testing components with html
- Replacing injectables in the injector tree
- Replacing directives in the component tree

Some of them do not have a good answer in AngularDart. In Angular2, where everything is hierarchical, they will be even harder to solve.

**Comment by mhevery** [comment]
I can take a stab at creating a design doc/ issues

**Comment by mhevery** [comment]
https://docs.google.com/document/d/1qFfHwYCOLpLuTap_l6TCEaU4I0C-pVFVbTl2o4-XbUo/edit

**Comment by vicb** [comment]
closed via #1160
